src/utilities/awala/InternetEndpoint.ts

```diff
@@ -83,7 +83,7 @@ export class InternetEndpoint extends Endpoint {
   public async saveChannel(
     connectionParams: PrivateEndpointConnParams,
     dbConnection: Connection,
-  ): Promise<void> {
+  ): Promise<InternetPrivateEndpointChannel> {
```

Comment:
This is fine, but it needs a test.

Reply:
There are 2 ways I could test this method with

```
    test('Result of save private endpoint channel method should returned', async () => {
      const spyOnSavePrivateEndpointChannel = jest.spyOn(endpoint, 'savePrivateEndpointChannel');
      const mockChannel = {
        peer: {
          id : 'test'
        }
      };
      spyOnSavePrivateEndpointChannel.mockResolvedValueOnce(mockChannel as any)

      const response = await endpoint.saveChannel(peerConnectionParams, dbConnection);
      
      expect(response).toStrictEqual(mockChannel);
    });
    ```
    
    
    ```
         const spyOnSavePrivateEndpointChannel = jest.spyOn(endpoint, 'savePrivateEndpointChannel');

      const response = await endpoint.saveChannel(peerConnectionParams, dbConnection);

      const savePrivateEndpointChannelResult = await spyOnSavePrivateEndpointChannel.mock.results[0].value;
      expect(response).toStrictEqual(savePrivateEndpointChannelResult);
      ```

Reply:
Remember what we've discussed about using spies/mocks with the endpoint or channel classes... Don't use them; check the outcome.